io/io/src/TGenCollectionProxy.cxx

```diff
@@ -941,7 +941,7 @@ TClass *TGenCollectionProxy::GetCollectionClass() const
 
 Int_t TGenCollectionProxy::GetCollectionType() const
 {
-   if (!fValue.load()) {
+   if (!fValue.load(std::memory_order_relaxed)) {
```

Comment:
NOTE: the use of `std::memory_order_relaxed` does not mean the non-atomic values set in `Initialize` will be properly synchronized across threads (assuming you were intending `fValue` to do that work).

Reply:
The intended pattern is that Initialize take the global lock, recheck fValue (and return immediately if set), do the work and set fValue as the last step.  (i.e. fValue being "spurriously" seen as nullptr is fine).